src/java/org/apache/cassandra/db/rows/UnfilteredRowIteratorWithLowerBound.java

```diff
@@ -185,70 +183,8 @@ public Row staticRow()
         return super.staticRow();
     }
 
-    private static <V> ClusteringBound<V> createArtificialLowerBound(boolean isReversed, ClusteringPrefix<V> from)
-    {
-        return from.accessor().factory().inclusiveOpen(isReversed, from.getRawValues()).artificialLowerBound(isReversed);
-    }
-
-    /**
-     * @return the lower bound stored on the index entry for this partition, if available.
-     */
-    private ClusteringBound<?> getKeyCacheLowerBound()
-    {
-        // NOTE: CASSANDRA-11206 removed the lookup against the key-cache as the IndexInfo objects are no longer
-        // in memory for not heap backed IndexInfo objects (so, these are on disk).
-        // CASSANDRA-11369 is there to fix this afterwards.
-
-        // Creating the iterator ensures that rowIndexEntry is loaded if available (partitions bigger than
-        // DatabaseDescriptor.column_index_size)
-        if (!canUseMetadataLowerBound())
-            maybeInit();
-
-        RowIndexEntry rowIndexEntry = sstable.getCachedPosition(partitionKey(), false);
```

Comment:
I think I have misled you here, checking the key cache still makes sense.

What I meant is that we do not want to initialize the iterator so that we can check the key cache to decide that we did not want to initialize the iterator. Also, calling `canUseMetadataLowerBound()` here is a bad idea in its own right.

I would remove the lines above this in the method, but still try it first in `lowerBound()`.

Reply:
Not you, the comment:
```
        // Creating the iterator ensures that rowIndexEntry is loaded if available (partitions bigger than
        // DatabaseDescriptor.column_index_size)
        if (!canUseMetadataLowerBound())
            maybeInit();
```